# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.cpp
# Commit: cb0e58368cd4
# Full Hash: cb0e58368cd4d8e1f79b193b7ccae9586cc2ab50
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 10: Use TaggedParserAtomIndex in StencilModuleEntry. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96917
# ==============================================================================

diff -r 73cc5fa0e0bd -r cb0e58368cd4 js/src/frontend/Parser.cpp
--- a/js/src/frontend/Parser.cpp	Sat Nov 14 00:03:20 2020 +0000
+++ b/js/src/frontend/Parser.cpp	Sat Nov 14 00:03:28 2020 +0000
@@ -1749,7 +1749,8 @@
   StencilModuleMetadata& moduleMetadata =
       this->compilationInfo_.stencil.moduleMetadata;
   for (auto entry : moduleMetadata.localExportEntries) {
-    const ParserAtom* nameId = entry.localName;
+    const ParserAtom* nameId =
+        this->compilationInfo_.stencil.getParserAtomAt(cx_, entry.localName);
     MOZ_ASSERT(nameId);
 
     DeclaredNamePtr p = modulepc.varScope().lookupDeclaredName(nameId);